radis/lbl/factory.py

```diff
@@ -911,10 +911,14 @@ def eq_spectrum_gpu(
         molpar = MolParams()
 
         try:
-            id_set = self.df0[
-                "id"
-            ].unique()  # get all the molecules in the dataframe, should ideally be 1 element for GPU
-            mol_id = id_set[0]
+            if "id" in self.df0:
+                id_set = self.df0[
+                    "id"
+                ].unique()  # get all the molecules in the dataframe, should ideally be 1 element for GPU
```

Comment:
>should ideally be 1 element for GPU
I think GPU can handle multiple species without too much problems, as long as we combine the mol_id and iso in a single identification array. For all intents and purposes, different isotopes might as well be diffent molecules.

Reply:
Mole fractions may be different : you still need a specific mol_id